packages/ckeditor5-engine/src/model/differ.js

```diff
@@ -249,23 +252,29 @@ export default class Differ {
 	 * Buffers a marker change.
 	 *
 	 * @param {String} markerName The name of the marker that changed.
-	 * @param {module:engine/model/range~Range|null} oldRange Marker range before the change or `null` if the marker has just
-	 * been created.
-	 * @param {module:engine/model/range~Range|null} newRange Marker range after the change or `null` if the marker was removed.
-	 * @param {Boolean} affectsData Flag indicating whether marker affects the editor data.
+	 * @param {module:engine/model/markercollection~MarkerData} oldMarkerData Marker range before the change
+	 * or `null` if the marker has just been created.
+	 * @param {module:engine/model/markercollection~MarkerData} newMarkerData Marker range after the change
+	 * or `null` if the marker was removed.
 	 */
-	bufferMarkerChange( markerName, oldRange, newRange, affectsData ) {
+	bufferMarkerChange( markerName, oldMarkerData, newMarkerData ) {
 		const buffered = this._changedMarkers.get( markerName );
 
 		if ( !buffered ) {
 			this._changedMarkers.set( markerName, {
-				oldRange,
-				newRange,
-				affectsData
+				oldRange: oldMarkerData.range,
+				newRange: newMarkerData.range,
+				affectsData: oldMarkerData.affectsData,
+				changesDataAffecting: oldMarkerData.affectsData !== newMarkerData.affectsData
```

Comment:
I didn't want to change this API now to limit the number of changes, but perhaps it would make sense to have the following one:

```typescript
{
	oldMarkerData: MarkerData;
	newMarkerData: MarkerData;
	changesDataAffecting: Boolean;
}
```

Or maybe even:

```typescript
{
	oldMarkerData: MarkerData;
	newMarkerData: MarkerData;
}
```

I was thinking that changing the `affectsData` twice should also produce the event, but it seems to be a super edge case (IDK if we have anywhere even a case of changing marker from affecting data to not affecting). So perhaps we can just check the first marker data from the first change with the last marker data from the last change for the same marker to determine changes.

Reply:
Fixed.